source/particle/property/grain_size.cc

```diff
@@ -66,39 +66,49 @@ namespace aspect
 
       template <int dim>
       void
-      GrainSize<dim>::update_particle_property(const unsigned int data_position,
-                                               const Vector<double> &solution,
-                                               const std::vector<Tensor<1,dim>> &gradients,
-                                               typename ParticleHandler<dim>::particle_iterator &particle) const
+      GrainSize<dim>::update_particle_properties(const unsigned int data_position,
+                                                 const std::vector<Vector<double>> &solution,
+                                                 const std::vector<std::vector<Tensor<1,dim>>> &gradients,
+                                                 typename ParticleHandler<dim>::particle_iterator_range &particles) const
       {
-        material_inputs.position[0] = particle->get_location();
-
-        material_inputs.current_cell = typename DoFHandler<dim>::active_cell_iterator(*particle->get_surrounding_cell(),
+        material_inputs  = MaterialModel::MaterialModelInputs<dim>(solution.size(), this->n_compositional_fields());
+        material_outputs = MaterialModel::MaterialModelOutputs<dim>(solution.size(), this->n_compositional_fields());
+        material_inputs.requested_properties = MaterialModel::MaterialProperties::reaction_terms;
+        material_inputs.current_cell = typename DoFHandler<dim>::active_cell_iterator(*particles.begin()->get_surrounding_cell(),
```

Comment:
Can it ever happen that someone passes a list of particles that are not all in one cell?